### PR TITLE
Create Magento's order even with failed payments

### PR DESCRIPTION
**Pull request by wallaceSF**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| **Issue**         | https://mundipagg.atlassian.net/browse/CONECT-1087
| **What?**         | Creates order and subscription independent the status, based on configuration.
| **Why?**          |Some customers wish to save all order creation attempts, successful or not
| **How?**          | Will be based on configuration, so as not to break the customers who do not want this feature

### CODE REVIEW

**Comment by wallaceSF** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 3
           

Complexity increasing per file
==============================
- src/Kernel/Factories/ConfigurationFactory.php  1
         

Clones added
============
- src/Kernel/Aggregates/Charge.php  1
- src/Recurrence/Aggregates/Charge.php  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/mundipagg/ecommerce-module-core/pullRequest?prid=5852032&bid=19156292)

File: src/Kernel/Aggregates/Configuration.php [hunk 1]
```diff
@@ -383,6 +401,14 @@ public function isSendMailEnabled()
         return $this->sendMailEnabled;
     }
 
+    /**
+     * @return bool
+     */
+    public function isCreateOrderEnabled()
```
Comment (GabrielDeveloper) [inline src/Kernel/Aggregates/Configuration.php]: colocar metodo como protected, para conseguir fazer a herança de multilojas

**Comment by wallaceSF** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 3
           

Complexity increasing per file
==============================
- src/Kernel/Factories/ConfigurationFactory.php  1
         

Clones added
============
- src/Kernel/Aggregates/Charge.php  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/mundipagg/ecommerce-module-core/pullRequest?prid=5852032&bid=19156292)

**Comment by wallaceSF** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 4
           

Complexity increasing per file
==============================
- src/Kernel/Factories/ConfigurationFactory.php  1
         

Clones added
============
- src/Kernel/Aggregates/Charge.php  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/mundipagg/ecommerce-module-core/pullRequest?prid=5852032&bid=19156292)

**Review by GabrielDeveloper — APPROVED** [review]

**Review by michelpl — APPROVED** [review]

**Pull request merged**